src/base/IoManip.h

```diff
@@ -76,27 +76,31 @@ operator <<(std::ostream &os, const RawPointerT<Pointer> &pd)
 }
 
 /// std::ostream manipulator to print integers as hex numbers prefixed by 0x
+/// Normally used through the asHex() convenience function
 template <class Integer>
 class AsHex
 {
 public:
-    explicit AsHex(const Integer n): io_manip(n) {}
+    explicit AsHex(const Integer n, const size_t w=0) : io_manip(n), width(w) {}
     Integer io_manip; ///< the integer to print
+    size_t width; ///< the width to use number of digits to print
 };
 
 template <class Integer>
 inline std::ostream &
 operator <<(std::ostream &os, const AsHex<Integer> number)
 {
     const auto oldFlags = os.flags();
-    os << std::hex << std::showbase << number.io_manip;
+    os << "0x" << std::hex <<
+        std::setfill('0') << std::setw(number.width) <<
+        number.io_manip;
     os.setf(oldFlags);
     return os;
 }
 
-/// a helper to ease AsHex object creation
+/// helpers to ease AsHex object creation
```

Comment:
Undo because there is only one helper:

```suggestion
/// a helper to ease AsHex object creation
```